### ISSUE TITLE
192-hugo-search

### ISSUE THREAD

**Issue by retifrav** [retifrav/retifrav.github.io#44]
[Search for a Hugo-based blog](https://retifrav.github.io/blog/2020/01/05/hugo-search/) comments.

**Comment by guzhiling** [comment]
Thank you for this post! I am very new to html and js, and found your posts extremely helpful.

Today, I tried to implement the search box on my side, but only the first result is returned when multiple matches are found, see the following image. 
![image](https://user-images.githubusercontent.com/37173670/84961129-1c6c5c00-b0c9-11ea-86af-01c8876c77d5.png)

I looked into the "search.js" and "partial/comments.html" there are no special format settings. I used hugo lithium, and I am wondering if you implemented some additional settings apart from those in the post. It would be great if you could give some hints.

(I also struggled to change the size of the input box, but this problem is very minor, and I think it may relate to the first question too. Please ignore the second question if you don't have time!)

Just in case it helps, my site repository is here https://github.com/guzhiling/guzhiling.com 

Thank you in advance, and please take your time to respond! :)

**Comment by retifrav** [comment]
## Failing search

I've checked your search, and indeed it shows only the first result, but if you check the console output, you'll see the following there:

``` console
Uncaught ReferenceError: Mark is not defined
```

And most likely what's going on is your script starts to iterate through search results, displays the first one, tries to use `mark.js` library to highlight keywords, but fails (as you did not include that library in your project) and stops iterating through the rest of search results.

So two possible fixes are: either add the library to your project or just comment out these lines (`new Mark(document.getElementById(...`), then your search should work:

<img width="841" alt="search" src="https://user-images.githubusercontent.com/6904927/84995628-88dc6f00-b14c-11ea-9c5d-ec3129721b7b.png">

## Search field size

I am not sure what size do you want it to have, so I'll just drop some suggestions here.

Firstly, I would add some padding to this input field so it looks a bit nicer:


``` css
#search-query {
    padding: 4px 8px;
}
```

Then, instead of manipulating styles inline, I would use classes for dormant/active search (because later you might want to switch back to dormant search when the field is empty and loses focus):

``` css
.search-query-dormant {
    display: inline-block;
    width: unset;
    margin: 0;
}
.search-query-active {
    display: block;
    width: 100%;
    margin: 5px 0;
}
```

As you can see here, instead of hardcoded `600px` I just set it to be a `block` with some vertical margin and to have 100% width. Not sure though if that's what you wanted to achieve.

And lastly, switch the classes using function:

``` html
<input id="search-query" class="search-query-dormant" placeholder="Search" onkeypress="activateSearch();" />
```
``` js
function activateSearch()
{
    document.getElementById("search-query").classList.remove("search-query-dormant");
    document.getElementById("search-query").classList.add("search-query-active");
}
```

**Comment by guzhiling** [comment]
Thank you so much for the detailed explanation. That solved all my problems!

**Comment by iootaa** [comment]
Looks great!
Did you document somewhere the way you highlight results?

**Comment by guzhiling** [comment]
Yes, the author commented 

```
// optionaly highlight the search query in excerpts 
// using mark.js (https://markjs.io) 

new Mark(document.getElementById("search-summary-" + key)) 
.mark(searchQuery, { "separateWordSearch": false });
```

**Comment by retifrav** [comment]
Yeah, this comment in the code is the only bit about highlighting. Since I based my post on another article, and highlighting is already mentioned there, I thought that should be enough.

But now I realized that I never mentioned including `mark.js`, so it may (and already did) confuse unprepared minds. Added a bit more details to the post.
